test/api/route/messages_test.dart

```diff
@@ -184,21 +191,51 @@ void main() {
         {'operator': 'stream', 'operand': 12},
         {'operator': 'topic', 'operand': 'stuff'},
       ]));
-      checkNarrow(const MentionsNarrow().apiEncode(), jsonEncode([
-        {'operator': 'is', 'operand': 'mentioned'},
+      checkNarrow(eg.topicNarrow(12, 'stuff', with_: 1).apiEncode(), jsonEncode([
+        {'operator': 'stream', 'operand': 12},
+        {'operator': 'topic', 'operand': 'stuff'},
+        {'operator': 'with', 'operand': 1},
       ]));
-      checkNarrow(const StarredMessagesNarrow().apiEncode(), jsonEncode([
-        {'operator': 'is', 'operand': 'starred'},
+      checkNarrow([ApiNarrowDm([123, 234])], jsonEncode([
+        {'operator': 'dm', 'operand': [123, 234]},
       ]));
 
-      checkNarrow([ApiNarrowDm([123, 234])], jsonEncode([
+      // Unlikely to occur in the wild but should still be handled correctly
+      checkNarrow([ApiNarrowDm([123, 234]), ApiNarrowWith(1)], jsonEncode([
         {'operator': 'dm', 'operand': [123, 234]},
+        {'operator': 'with', 'operand': 1},
       ]));
 
       connection.zulipFeatureLevel = 176;
+
+      checkNarrow(eg.topicNarrow(12, 'stuff', with_: 1).apiEncode(), jsonEncode([
```

Comment:
nit: and then without those other comments, the change to the ZFL can be comfortably kept in the same stanza with the checks that rely on it
```suggestion
      ]));

      connection.zulipFeatureLevel = 176;
      checkNarrow(eg.topicNarrow(12, 'stuff', with_: 1).apiEncode(), jsonEncode([
```

I think that substantially helps to mitigate the risk of confusion from this use of mutable state, where one otherwise might see one of those later `checkNarrow` calls in isolation and wonder why the expected behavior is what it is